lib/internal/Magento/Framework/HTTP/Adapter/Curl.php

```diff
@@ -175,6 +175,9 @@ public function write($method, $url, $http_ver = '1.1', $headers = [], $body = '
         if ($method == \Zend_Http_Client::POST) {
             curl_setopt($this->_getResource(), CURLOPT_POST, true);
             curl_setopt($this->_getResource(), CURLOPT_POSTFIELDS, $body);
+        } elseif ($method == \Zend_Http_Client::PUT) {
+            curl_setopt($this->_getResource(), CURLOPT_CUSTOMREQUEST, 'PUT');
```

Comment:
I stand corrected as to the deprecation, but in that case the recommendation is to use `CURLOPT_UPLOAD`, it will automatically assume the HTTP `PUT` method unless otherwise specified, and then you then still have the ability to handle `CURLOPT_INFILESIZE` or `CURLOPT_INFILESIZE_LARGE`.